### ISSUE TITLE
[feature request]add es6 Decorators.

### ISSUE THREAD

**Issue by sonygod** [haxiomic/dts2hx#24]
how about consider  es6 Decorators in Haxe.

[link0](https://www.typescriptlang.org/docs/handbook/decorators.html)

because cocoa creator use that for beautiful code custom component definition.

**Comment by haxiomic** [comment]
I agree this would be nice and it's certainly possible with a compiler macro but quite a lot of work

I don't plan to do this in the short term. If someone else makes a es6 decorator library, then we could explore integrating it into dts2hx
